python-avd/pyavd/get_fabric_documentation.py

```diff
@@ -166,31 +167,39 @@ def _act_dynamic_digital_twin_fabric_documentation(
 
 
 def _get_digital_twin_act(fabric_documentation_facts: FabricDocumentationFacts) -> ACTDigitalTwin:
-    digital_twin_node_types: set[str] = set()
-
     # Identify common username for fabric nodes
     # Value is enforced as a non-empty string during the generation of the metadata part of the structured_config
     digital_twin_fabric_username: str = next(
         (
-            get(device_structurude_config, "metadata.digital_twin.username")
-            for device_structurude_config in fabric_documentation_facts.structured_configs.values()
+            get(device_structured_config, "metadata.digital_twin.username")
+            for device_structured_config in fabric_documentation_facts.structured_configs.values()
         ),
     )
 
     # Identify common password for fabric nodes
     # Value is enforced as a non-empty string during the generation of the metadata part of the structured_config
     digital_twin_fabric_password: str = next(
         (
-            get(device_structurude_config, "metadata.digital_twin.password")
-            for device_structurude_config in fabric_documentation_facts.structured_configs.values()
+            get(device_structured_config, "metadata.digital_twin.password")
+            for device_structured_config in fabric_documentation_facts.structured_configs.values()
         ),
     )
 
+    digital_twin_node_types: dict[str, ActNodeTypeSettings | None] = {
+        "cloudeos": None,
+        "cvp": None,
+        "generic": None,
+        "third_party": None,
+        "tools_server": None,
+        "veos": None,
+    }
     digital_twin_devices: list[dict[str, ActNodeSettings]] = []
     device_list: list[str] = list(fabric_documentation_facts.avd_facts)
     for device in sorted(device_list):
-        if (digital_twin_node_type := get(fabric_documentation_facts.structured_configs, f"{device}.metadata.digital_twin.node_type")) is not None:
-            digital_twin_node_types.add(digital_twin_node_type)
+        if (
+            digital_twin_node_type := get(fabric_documentation_facts.structured_configs, f"{device}.metadata.digital_twin.node_type")
+        ) in digital_twin_node_types and not digital_twin_node_types[digital_twin_node_type]:
```

Comment:
We are skipping here node_types which have "-" in their names (third-party and tools-server)

```suggestion
            digital_twin_node_type := get(fabric_documentation_facts.structured_configs, f"{device}.metadata.digital_twin.node_type", "").replace("-", "_")
        ) in digital_twin_node_types and not digital_twin_node_types[digital_twin_node_type]:
```

Reply:
I modified the keys of the temp dict instead.